tests/test_spatial_avg.py

```diff
@@ -453,6 +453,46 @@ def test_returns_area_weights_for_region_within_lon(self):
 
         xr.testing.assert_allclose(result, expected)
 
+
+class TestGetLongitudeWeights:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.ds = generate_dataset(cf_compliant=True, has_bounds=True)
+
+    def test_returns_area_weights_for_region_within_lon(self):
+        expected = xr.DataArray(
+            data=np.array([0.0, 0.0, 50.0, 0.0]),
+            coords={"lon": self.ds.lon},
+            dims=["lon"],
+        )
+        # Longitude axes orientation swaps from (-180, 180) to (0, 360).
+        result = self.ds.spatial._get_longitude_weights(
+            domain_bounds=self.ds.lon_bnds.copy(),
+            region_bounds=np.array([-170.0, -120.0]),
+        )
+
+        xr.testing.assert_allclose(result, expected)
+
+    def test_returns_area_weights_for_region_within_lon_including_prime_meridian_cell(
+        self,
+    ):
+        ds = self.ds.copy()
+        ds.lon_bnds.data[:] = np.array([[359, 1], [1, 90], [90, 180], [180, 359]])
+
+        expected = xr.DataArray(
+            data=np.array([0.0, 0.0, 0.0, 50.0]),
+            coords={"lon": self.ds.lon},
+            dims=["lon"],
+        )
+
+        # Longitude axes orientation swaps from (-180, 180) to (0, 360).
+        result = self.ds.spatial._get_longitude_weights(
+            domain_bounds=self.ds.lon_bnds,
+            region_bounds=np.array([-170.0, -120.0]),
+        )
+
+        xr.testing.assert_allclose(result, expected)
```

Comment:
Is this test doing what is suggested in the name? It seems like it is supposed to specify a region that includes grid cells that span prime meridian (e.g., the cell from 359 to 1), but it specifies a domain of -170 to -120 (equivalently 190 to 240).

Reply:
I see the confusion in the name of the test. It should be called `def test_returns_area_weights_for_lon_domain_spanning_prime_meridian_scaled_to_region`.

Essentially, the domain has a prime meridian cell and it is scaled down to the region (190 to 240).

> It seems like it is supposed to specify a region that includes grid cells that span prime meridian (e.g., the cell from 359 to 1)

I'll add this test case too.